### ISSUE TITLE
Bad cast from type DB::ColumnNothing to DB::ColumnString assertion failed

### ISSUE THREAD

**Issue by nikitamikhaylov** [ClickHouse/ClickHouse#13917]
```sql
SELECT [100, NULL, 1048575, 0.5, NULL, NULL], has(materialize(['', NULL, '', '', '']), materialize(NULL))
```
```bash
2020.08.20 14:48:08.253558 [ 35345 ] {9e364f20-0959-4230-abea-beff566d9286} <Error> : Logical error: 'Bad cast from type DB::ColumnNothing to DB::ColumnString'.
clickhouse: ../src/Common/Exception.cpp:45: DB::Exception::Exception(const std::string &, int): Assertion `false' failed.
2020.08.20 14:48:08.254004 [ 35241 ] {} <Trace> BaseDaemon: Received signal 6
2020.08.20 14:48:08.254358 [ 35502 ] {} <Fatal> BaseDaemon: ########################################
2020.08.20 14:48:08.254717 [ 35502 ] {} <Fatal> BaseDaemon: (version 20.8.1.1, build id: 8924309CF00424A4) (from thread 35345) (query_id: 9e364f20-0959-4230-abea-beff566d9286) Received signal Aborted (6)
2020.08.20 14:48:08.254923 [ 35502 ] {} <Fatal> BaseDaemon: 
2020.08.20 14:48:08.255094 [ 35502 ] {} <Fatal> BaseDaemon: Stack trace: 0x7fda15ec218b 0x7fda15ea1859 0x7fda15ea1729 0x7fda15eb2f36 0x183349f1 0x1859d444 0x1f96ae60 0x1d987f5c 0x1d987425 0x1d8fcb77 0x1d8fa51a 0x1d8f9d7d 0x1993540f 0x1999cfcb 0x1999e0ad 0x1ffa9a41 0x1ffad1d3 0x210fdb9c 0x207a9d55 0x20e722a8 0x20ed236c 0x20ed22cf 0x20ed228d 0x20ed223d 0x20ed220d 0x20ed136e 0x18357735 0x18356655
2020.08.20 14:48:08.256857 [ 35502 ] {} <Fatal> BaseDaemon: 4. /build/glibc-YYA7BZ/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51: __GI_raise @ 0x4618b in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.31.so
2020.08.20 14:48:08.257419 [ 35502 ] {} <Fatal> BaseDaemon: 5. /build/glibc-YYA7BZ/glibc-2.31/stdlib/abort.c:81: abort @ 0x25859 in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.31.so
2020.08.20 14:48:08.258151 [ 35502 ] {} <Fatal> BaseDaemon: 6. /build/glibc-YYA7BZ/glibc-2.31/intl/loadmsgcat.c:509: _nl_load_domain.cold @ 0x25729 in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.31.so
2020.08.20 14:48:08.258492 [ 35502 ] {} <Fatal> BaseDaemon: 7. ? @ 0x36f36 in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.31.so
2020.08.20 14:48:08.258754 [ 35502 ] {} <Fatal> BaseDaemon: 8. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Common/Exception.cpp:48: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x183349f1 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.260954 [ 35502 ] {} <Fatal> BaseDaemon: 9. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Common/assert_cast.h:47: DB::ColumnString const& assert_cast<DB::ColumnString const&, DB::IColumn const&>(DB::IColumn const&) @ 0x1859d444 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.292489 [ 35502 ] {} <Fatal> BaseDaemon: 10. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Columns/ColumnString.h:223: DB::ColumnString::compareAt(unsigned long, unsigned long, DB::IColumn const&, int) const @ 0x1f96ae60 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.317481 [ 35502 ] {} <Fatal> BaseDaemon: 11. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/array/arrayIndex.h:511: DB::ArrayIndexGenericImpl<DB::IndexToOne, false>::vectorCase4(DB::IColumn const&, DB::PODArray<unsigned long, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, DB::IColumn const&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul>&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) @ 0x1d987f5c in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.342336 [ 35502 ] {} <Fatal> BaseDaemon: 12. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/array/arrayIndex.h:0: DB::ArrayIndexGenericImpl<DB::IndexToOne, false>::vector(DB::IColumn const&, DB::PODArray<unsigned long, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, DB::IColumn const&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul>&, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const*, DB::PODArray<char8_t, 4096ul, Allocator<false, false>, 15ul, 16ul> const*) @ 0x1d987425 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.362049 [ 35502 ] {} <Fatal> BaseDaemon: 13. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/array/arrayIndex.h:841: DB::FunctionArrayIndex<DB::IndexToOne, DB::NameHas>::executeGeneric(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long) const @ 0x1d8fcb77 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.381610 [ 35502 ] {} <Fatal> BaseDaemon: 14. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/array/arrayIndex.h:990: DB::FunctionArrayIndex<DB::IndexToOne, DB::NameHas>::perform(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long) const @ 0x1d8fa51a in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.401617 [ 35502 ] {} <Fatal> BaseDaemon: 15. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/array/arrayIndex.h:977: DB::FunctionArrayIndex<DB::IndexToOne, DB::NameHas>::executeImpl(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) const @ 0x1d8f9d7d in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.406696 [ 35502 ] {} <Fatal> BaseDaemon: 16. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/IFunctionAdaptors.h:152: DB::DefaultExecutable::execute(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long) @ 0x1993540f in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.411046 [ 35502 ] {} <Fatal> BaseDaemon: 17. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/IFunction.cpp:326: DB::ExecutableFunctionAdaptor::executeWithoutLowCardinalityColumns(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) @ 0x1999cfcb in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.415358 [ 35502 ] {} <Fatal> BaseDaemon: 18. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Functions/IFunction.cpp:486: DB::ExecutableFunctionAdaptor::execute(DB::Block&, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, unsigned long, unsigned long, bool) @ 0x1999e0ad in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.445813 [ 35502 ] {} <Fatal> BaseDaemon: 19. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Interpreters/ExpressionActions.cpp:365: DB::ExpressionAction::execute(DB::Block&, bool) const @ 0x1ffa9a41 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.476374 [ 35502 ] {} <Fatal> BaseDaemon: 20. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Interpreters/ExpressionActions.cpp:684: DB::ExpressionActions::execute(DB::Block&, bool) const @ 0x1ffad1d3 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.515426 [ 35502 ] {} <Fatal> BaseDaemon: 21. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Processors/Transforms/ExpressionTransform.cpp:41: DB::ExpressionTransform::transform(DB::Chunk&) @ 0x210fdb9c in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.551551 [ 35502 ] {} <Fatal> BaseDaemon: 22. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Processors/ISimpleTransform.h:43: DB::ISimpleTransform::transform(DB::Chunk&, DB::Chunk&) @ 0x207a9d55 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.589600 [ 35502 ] {} <Fatal> BaseDaemon: 23. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Processors/ISimpleTransform.cpp:89: DB::ISimpleTransform::work() @ 0x20e722a8 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.627165 [ 35502 ] {} <Fatal> BaseDaemon: 24. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Processors/Executors/PipelineExecutor.cpp:78: DB::executeJob(DB::IProcessor*) @ 0x20ed236c in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.664775 [ 35502 ] {} <Fatal> BaseDaemon: 25. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../src/Processors/Executors/PipelineExecutor.cpp:95: DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0::operator()() const @ 0x20ed22cf in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.703204 [ 35502 ] {} <Fatal> BaseDaemon: 26. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/type_traits:3519: decltype(std::__1::forward<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(fp)()) std::__1::__invoke<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&) @ 0x20ed228d in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.741536 [ 35502 ] {} <Fatal> BaseDaemon: 27. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/__functional_base:349: void std::__1::__invoke_void_return_wrapper<void>::__call<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&) @ 0x20ed223d in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.780169 [ 35502 ] {} <Fatal> BaseDaemon: 28. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/functional:1540: std::__1::__function::__alloc_func<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0, std::__1::allocator<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0>, void ()>::operator()() @ 0x20ed220d in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.818795 [ 35502 ] {} <Fatal> BaseDaemon: 29. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/functional:1714: std::__1::__function::__func<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0, std::__1::allocator<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0>, void ()>::operator()() @ 0x20ed136e in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.819258 [ 35502 ] {} <Fatal> BaseDaemon: 30. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/functional:1867: std::__1::__function::__value_func<void ()>::operator()() const @ 0x18357735 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
2020.08.20 14:48:08.819909 [ 35502 ] {} <Fatal> BaseDaemon: 31. /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/../contrib/libcxx/include/functional:2473: std::__1::function<void ()>::operator()() const @ 0x18356655 in /home/jakalletti/ClickHouse/ClickHouse/build-clang-debug/programs/clickhouse
```
From https://clickhouse-test-reports.s3.yandex.net/13624/00fde75e2bd00a9d1fcba81c5442f07d5aab3612/fuzzer/fuzzer.log#fail1

**Comment by alexey-milovidov** [comment]
Should be fixed in #12550 ?

**Comment by myrrc** [comment]
> Should be fixed in #12550 ?

Yup.
